src/ibase.rs

```diff
@@ -2474,11 +2474,11 @@ pub struct XSQLDA {
     pub sqldabc: ISC_LONG,
     pub sqln: ISC_SHORT,
     pub sqld: ISC_SHORT,
-    pub sqlvar: [XSQLVAR; 32usize],
+    pub sqlvar: [XSQLVAR; 1usize],
```

Comment:
This change was making the length calculation alloc more than necessary, because mem::size_of<XSQLDA>() was returning a lenght of a XSQLDA with space for 32 fields of XSQLVAR instead of just one as the c macro implied.